capstone/src/main/webapp/script.js

```diff
@@ -31,14 +32,160 @@ async function submitInput() {
   const attributes = document.getElementById("available-attributes").getElementsByTagName('input');
-  const requestedAttributes = []
+  const requestedAttributes = [];
   for (const attribute of attributes) {
     if (attribute.checked) {
       requestedAttributes.push(attribute.value);	
     }	
   }
-  await callPerspective(textElement.value, langElement.value, requestedAttributes);
+
+  // Get the selected analysis type
+  document.getElementById('analysis-container').innerHTML = '';
+  const radios = document.getElementsByName('analysisRadios');
+  var tokenizer = "";
+  for (i = 0; i < radios.length; i++) {
+    if (radios[i].checked) {
+      tokenizer = radios[i].value;
+      break;
+    }
+  }
+
+  handleInput(textElement.value, langElement.value, requestedAttributes, tokenizer);
+}
+
+/** Submits the input to Perspective and loads the appropriate output */
+async function handleInput(text, lang, requestedAttributes, tokenizer) {
+  // Draw the separating line for the output
+  const separator = document.getElementById('separator-container');
+  separator.innerHTML = '';
+  separator.appendChild(document.createElement('hr')); 
+	
+  // Make Perspective call for the entire submission and load graph data
+  const toxicityData = await callPerspective(text, lang, requestedAttributes);
+  loadChartsApi(toxicityData);
+
+  // Get detailed analysis if requested
+  if (tokenizer != "") {
+    getAnalysis(text, lang, requestedAttributes, tokenizer); 
+  }
+}
+
+/** Prints detailed analysis of text by word or sentence */
+async function getAnalysis(text, lang, requestedAttributes, tokenizer) {
+  // Set up the detailed analysis section
+  const analysisContainer = document.getElementById('analysis-container');
+  analysisContainer.appendChild(createAnyElement('b', 'Detailed Anaylsis'));
+  const loadingEl = document.createElement('p');
+  loadingEl.className = 'spinner-border';
+  analysisContainer.appendChild(loadingEl);
+  const result = document.createElement('p');
+  result.className = 'detailed-analysis';
+  
+  // Generate the results for every substring of the input text
+  const substrings = getSubstrings(text, tokenizer);
+  for (i = 0; i < substrings.length; i++) {
+    if (substrings[i] != '') {
+      // Get the Perspective scores for the substring & sort them descending
+      const response = await callPerspective(substrings[i], lang, requestedAttributes);
+      if (typeof(response.error) != 'undefined') {
+        analysisContainer.removeChild(loadingEl);
+        analysisContainer.appendChild(createAnyElement('p', 'Perspective API was not able to get scores for detailed analysis'));
+        return;
+      }
+      const toxicityScore = response.attributeScores.TOXICITY.summaryScore.value;
+      const attributes = sortAttributes(response);
+
+      // Color the substring appropriately	
+      const substringEl = createAnyElement('span', substrings[i]);
+			colorSubstring(substringEl, toxicityScore);
```

Comment:
nit: Let's fix the indentation on this line.

Reply:
Fixed